pkg/controllers/readiness/wellknown_ready_controller.go

```diff
@@ -117,47 +120,49 @@ func (c *wellKnownReadyController) sync(ctx context.Context, controllerContext f
 	// the well-known endpoint cannot be ready until we know the oauth-server's hostname
 	_, err = c.routeLister.Routes("openshift-authentication").Get("oauth-openshift")
 	if apierrors.IsNotFound(err) {
-		statusUpdates = append(statusUpdates, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
-			Type:    "WellKnownAvailable",
-			Status:  operatorv1.ConditionFalse,
-			Reason:  "PrereqsNotReady",
-			Message: err.Error(),
-		}))
+		status = status.WithConditions(applyoperatorv1.OperatorCondition().
+			WithType("WellKnownAvailable").
+			WithStatus(operatorv1.ConditionFalse).
+			WithReason("PrereqsNotReady").
+			WithMessage(err.Error()))
 	}
 	if err != nil {
 		return err
 	}
 
 	if err := c.isWellknownEndpointsReady(ctx, operatorSpec, operatorStatus, authConfig, infraConfig); err != nil {
-		statusUpdates = append(statusUpdates, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
-			Type:    "WellKnownAvailable",
-			Status:  operatorv1.ConditionFalse,
-			Reason:  "NotReady",
-			Message: fmt.Sprintf("The well-known endpoint is not yet available: %s", err.Error()),
-		}))
+		status = status.WithConditions(applyoperatorv1.OperatorCondition().
```

Comment:
`WithConditions` appends, so you can have two `WellKnownAvailable`.

Perhaps it's better to have separate method variables for each condition we're going to set and then put them together in defer?

Reply:
Good idea, created individual variables for conditions and added them to the status in defer.

Reply:
I removed the changes for this controller and will handle them in a separate PR (https://github.com/openshift/cluster-authentication-operator/pull/705).